src/main/java/io/confluent/idesidecar/restapi/models/graph/RealCCloudFetcher.java

```diff
@@ -534,4 +539,80 @@ protected CCloudOAuthContext.OrganizationDetails currentOrgDetails(String connec
     }
     return null;
   }
+
+  public Multi<CCloudFlinkComputePool> getFlinkComputePools(
+      String connectionId,
+      String envId
+  ) {
+    var headers = headersFor(connectionId);
+    String url = CONFLUENT_CLOUD_FLINK_COMPUTE_POOLS_URI.formatted(envId);
+    Log.infof("Fetching Flink compute pools from URL: %s with headers: %s", url, headers);
+    return listItems(headers, url, null, this::parseFlinkComputePoolsList)
+        .map(pool -> pool.withConnectionId(connectionId));
+  }
+
+  private PageOfResults<CCloudFlinkComputePool> parseFlinkComputePoolsList(
+      String json,
+      PaginationState state
+  ) {
+    return parseList(json, state, ListFlinkComputePoolsResponse.class);
+  }
+
+  @RegisterForReflection
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  private record ListFlinkComputePoolsResponse(
+      @JsonProperty(value = "api_version") String apiVersion,
+      String kind,
+      ListMetadata metadata,
+      @JsonProperty(value = "data", required = true) List<FlinkComputePoolResponse> data
+  ) implements ListResponse<FlinkComputePoolResponse, CCloudFlinkComputePool> {
+
+  }
+
+  @RegisterForReflection
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  private record FlinkComputePoolResponse(
+      @JsonProperty(value = "api_version") String apiVersion,
+      String kind,
+      @JsonProperty(required = true) String id,
+      JsonNode metadata,
+      @JsonProperty(value="environment") CCloudReference environment,
+      @JsonProperty(value="organization") CCloudReference organization,
```

Comment:
Hm, I think these fields are not part of the API spec for the entries of the compute list's `data` field.

Reply:
oh, and we wouldn't need them at this level anyway, thanks!